src/ir/storage_test.cc

```diff
@@ -0,0 +1,52 @@
+//-----------------------------------------------------------------------------
+// Copyright 2021 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     https://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//----------------------------------------------------------------------------
+#include "src/ir/storage.h"
+
+#include "src/common/testing/gtest.h"
+#include "src/ir/types/entity_type.h"
+#include "src/ir/types/type.h"
+#include "src/ir/types/type_factory.h"
+
+namespace raksha::ir {
+namespace {
+
+class StorageTest : public testing::Test {
+ protected:
+  types::TypeFactory type_factory_;
+};
+
+TEST_F(StorageTest, PropertyAccessorsReturnCorrectValues) {
+  Storage input_storage("input", type_factory_.MakePrimitiveType());
+  EXPECT_EQ(input_storage.name(), "input");
+  EXPECT_EQ(input_storage.type().type_base().kind(),
+            types::TypeBase::Kind::kPrimitive);
+
+  const types::Schema& schema = type_factory_.RegisterSchema("TestStruct", {});
+  Storage output_storage("output", type_factory_.MakeEntityType(schema));
+  EXPECT_EQ(output_storage.name(), "output");
+  const types::TypeBase& type_base = output_storage.type().type_base();
+  EXPECT_EQ(type_base.kind(), types::TypeBase::Kind::kEntity);
+  EXPECT_EQ(static_cast<const types::EntityType&>(type_base).schema().name(),
+            "TestStruct");
```

Comment:
If this were to fail due to `type_base` not being downcastable to `EntityType`, we would get a segfault due to dereferencing `nullptr`. Perhaps it would be better to split this into a statement that does the downcast, then `ASSERT_NOT_NULL` (or whether the appropriate assertion is) then the `EXPECT_EQ`.

Reply:
I intended to capture it by the check `EXPECT_EQ(type_base.kind(), types::TypeBase::Kind::kEntity)` in the previous line? However, I think you are right that that previous check should actually be `ASSERT_EQ` so that we don't segfault on `42`. Will change it.